numpy/lib/function_base.py

```diff
@@ -1897,24 +1899,25 @@ def diff(a, n=1, axis=-1):
     if n < 0:
         raise ValueError(
             "order must be non-negative but got " + repr(n))
+
     a = asanyarray(a)
     nd = a.ndim
-    slice1 = [slice(None)]*nd
-    slice2 = [slice(None)]*nd
-    slice1[axis] = slice(1, None)
-    slice2[axis] = slice(None, -1)
-    slice1 = tuple(slice1)
-    slice2 = tuple(slice2)
+    rng = range(nd)
+    axis = rng[axis]
```

Comment:
I took a look at the [implementation](https://github.com/numpy/numpy/blob/260f18c910175bcb6162feb363112929653276aa/numpy/core/src/multiarray/common.h#L142), and I seriously doubt that creating and indexing a `range` has any chance of being faster.